streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamNewProcessorApiTest.java

```diff
@@ -0,0 +1,117 @@
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+import static java.util.Arrays.asList;
+
+
+public class KStreamNewProcessorApiTest {
+
+    @Test
+    @DisplayName("Test for using new Processor API and state stores with the DSL")
+    void shouldGetStateStoreWithNewProcessor() {
+        final StreamsBuilder builder = new StreamsBuilder();
+        final StoreBuilder<?> storeBuilder = Stores.keyValueStoreBuilder(Stores.inMemoryKeyValueStore("store"), Serdes.String(), Serdes.String());
+
+
+        builder.stream("input", Consumed.with(Serdes.String(), Serdes.String()))
+                .processValues(new TransformerSupplier(storeBuilder), "store")
+                .to("output", Produced.with(Serdes.String(), Serdes.String()));
+
+        final List<KeyValue<String, String>> words = Arrays.asList(KeyValue.pair("a", "foo"), KeyValue.pair("b", "bar"), KeyValue.pair("c", "baz"));
+        try (TopologyTestDriver testDriver = new TopologyTestDriver(builder.build())) {
+            final TestInputTopic<String, String>
+                    testDriverInputTopic =
+                    testDriver.createInputTopic("input", Serdes.String().serializer(), Serdes.String().serializer());
+
+            words.forEach(clk -> testDriverInputTopic.pipeInput(clk.key, clk.value));
+
+            final List<String> expectedOutput = asList("fooUpdated", "barUpdated", "bazUpdated");
+
+            final Deserializer<String> keyDeserializer = Serdes.String().deserializer();
+            final List<String> actualOutput =
+                    new ArrayList<>(testDriver.createOutputTopic("output", keyDeserializer, Serdes.String().deserializer()).readValuesToList());
+
+            Assertions.assertEquals(expectedOutput, actualOutput);
+        }
+    }
+    private static class TransformerSupplier implements FixedKeyProcessorSupplier<String, String, String> {
+        private final StoreBuilder<?> storeBuilder;
+
+        public TransformerSupplier(final StoreBuilder<?> storeBuilder) {
+            this.storeBuilder = storeBuilder;
+        }
+
+        @Override
+        public ContextualFixedKeyProcessor<String, String, String> get() {
+            return new ContextualFixedKeyProcessor<String, String, String>() {
+                KeyValueStore<String, String> store;
+                FixedKeyProcessorContext<String, String> context;
+
+                @Override
+                public void init(final FixedKeyProcessorContext<String, String> context) {
+                    super.init(context);
+                    store = context.getStateStore("store");
+                    this.context = context;
+                }
+
+                @Override
+                public void process(final FixedKeyRecord<String, String> record) {
+                    final String updated = store.get(record.key());
+                    store.putIfAbsent(record.key(), record.value() + "Updated");
```

Comment:
Seems the only test guard is here, ie, if `store == null` we would crash? Should the test code above actually verify the store content in addition to the output topic content?

Reply:
ack - will test the store content as well